proposals/csharp-9.0/records.md

```diff
@@ -197,6 +197,118 @@ If the containing record is abstract, the synthesized clone method is also abstr
 If the "clone" method is not abstract, it returns the result of a call to a copy constructor. 
 
 
+### Printing members: PrintMembers and ToString
+
+If the record is derived from `object`, the record includes a synthesized method equivalent to a method declared as follows:
+```C#
+protected virtual void PrintMembers(System.StringBuilder builder, bool includeSeparator);
+```
+The method is `virtual` and `protected`.
+It is an error if the method is declared explicitly.
+
+The method:
+1. appends a separator ", " to `builder` if `includeSeparator` is true and the record has public fields or properties,
+2. for each of the record's public field and property members, appends that member's name followed by " = " followed by the result of invoking `object.ToString()` on that member's value: `this.member.ToString()`, separated with ", ",
+
+If the record type is derived from a base record `Base`, the record includes a synthesized override equivalent to a method declared as follows:
+```C#
+protected override void PrintMembers(StringBuilder builder, bool includeSeparator);
+```
+It is an error if the override is declared explicitly.
+
+The method appends the same contents to `builder` as described above, but also then calls `base.PrintMembers` with two arguments:
+1. its `builder` parameter,
+2. a true `includeSeparator` argument if either (a) the record has public fields or properties, or (b) its `includeSeparator` parameter was given as true.
+
+The record includes a synthesized method equivalent to a method declared as follows:
+```C#
+public override string ToString();
+```
+
+The method can be declared explicitly. It is an error if the explicit declaration does not match the expected signature or accessibility, or if the explicit declaration doesn't allow overiding it in a derived type and the record type is not `sealed`. It is an error if the method doesn't override a method with this signature in the record `Base` (for example, if the method is missing in the `Base`, or sealed, or not virtual, etc.).
+
+The synthesized method:
+1. creates a `StringBuilder` instance,
+2. appends the record name to the builder, followed by " { ",
```

Comment:
FQN, or just name?
What name is added if this is a generic record?  do we include type parameeters?

Reply:
just name, without type parameters.